### PR TITLE
chore: add a default to setup to create an admin keycloak user

### PR DESCRIPTION
**Pull request by Racer159**
This adds the option to automatically create a keycloak admin user

### CODE REVIEW

File: tasks/setup.yaml [hunk 1]
```diff
@@ -5,16 +5,72 @@ tasks:
...
+
+          # Create the doug user in the UDS Realm
+          curl --location "https://keycloak.admin.uds.dev/admin/realms/uds/users" \
+          --header "Content-Type: application/json" \
+          --header "Authorization: Bearer ${KEYCLOAK_ADMIN_TOKEN}" \
+          --data-raw '{
+              "username": "doug",
+              "firstName": "Doug",
+              "lastName": "Unicorn",
+              "email": "doug@defenseunicorns.com",
```
Comment (zachariahmiller) [inline tasks/setup.yaml:53]:
```suggestion
              "email": "doug@uds.dev",
```

I'd assume this makes more sense, but 🤷

File: tasks/setup.yaml [hunk 2]
```diff
@@ -5,16 +5,72 @@ tasks:
...
+    actions:
+      - description: Create a user named 'doug' in the uds realm of keycloak (using the default admin account)
+        cmd: |
+          KEYCLOAK_ADMIN_PASSWORD=$(./uds zarf tools kubectl get secret -n keycloak keycloak-admin-password -o jsonpath={.data.password} | base64 -d)
+          KEYCLOAK_ADMIN_TOKEN=$(curl -s --location "https://keycloak.admin.uds.dev/realms/master/protocol/openid-connect/token" \
+            --header "Content-Type: application/x-www-form-urlencoded" \
+            --data-urlencode "username=admin" \
+            --data-urlencode "password=${KEYCLOAK_ADMIN_PASSWORD}" \
+            --data-urlencode "client_id=admin-cli" \
+            --data-urlencode "grant_type=password" | yq .access_token)
```
Comment (zachariahmiller) [inline tasks/setup.yaml:43]:
```suggestion
            --data-urlencode "grant_type=password" | ./ uds zarf tools yq .access_token)
```

I guess im not certain if we can do this yet. If not please ignore.

File: tasks/setup.yaml [hunk 3]
```diff
@@ -5,16 +5,72 @@ tasks:
...
+                  "type": "password",
+                  "value": "unicorn123!@#",
+                  "temporary": false
+                }
+              ]
+          }'
+
+          # Disable 2FA
+          CONDITIONAL_OTP_ID=$(curl --location "https://keycloak.admin.uds.dev/admin/realms/uds/authentication/flows/Authentication/executions" \
+            --header "Authorization: Bearer ${KEYCLOAK_ADMIN_TOKEN}" | yq '.[] | select(.displayName == "Conditional OTP") | .id')
```
Comment (zachariahmiller) [inline tasks/setup.yaml:68]:
```suggestion
            --header "Authorization: Bearer ${KEYCLOAK_ADMIN_TOKEN}" | ./uds zarf tools yq '.[] | select(.displayName == "Conditional OTP") | .id')
```
same comment as the other one

**Review by zachariahmiller — CHANGES_REQUESTED** [review]
Not sure on the embedded `uds zarf yq` availability yet, but the email should probably change 🤔

**Review by zachariahmiller — APPROVED** [review]
:shipit:

**Comment by UncleGedd** [comment]
You glorious 🦄 s. Here I am 6 months later wondering how I'm going test SSO in UDS Runtime, and I find this 🤌

**Pull request merged**